mllib/src/test/scala/org/apache/spark/ml/recommendation/ALSSuite.scala

```diff
@@ -520,6 +488,81 @@ class ALSSuite
   }
 }
 
+class ALSCleanerSuite extends SparkFunSuite {
+  test("Clean shuffles") {
+    val conf = new SparkConf()
+    val localDir = Utils.createTempDir()
+    val tempDir = Utils.createTempDir()
+    def getAllFiles: Set[File] =
+      FileUtils.listFiles(localDir, TrueFileFilter.INSTANCE, TrueFileFilter.INSTANCE).asScala.toSet
+    try {
+      conf.set("spark.local.dir", localDir.getAbsolutePath)
+      conf.set("spark.shuffle.manager", "sort")
+      val sc = new SparkContext("local[2]", "test", conf)
+      try {
+        sc.setCheckpointDir(tempDir.getAbsolutePath)
+        // Test checkpoint and clean parents
+        val filesBefore = getAllFiles
+        val input = sc.parallelize(1 to 1000)
+        val keyed = input.map(x => (x % 20, 1))
+        val shuffled = keyed.reduceByKey(_ + _)
+        val keysOnly = shuffled.map{case (x, _) => x}
+        val deps = keysOnly.dependencies
+        keysOnly.checkpoint()
```

Comment:
Is it necessary to include `checkpoint` in the test of "ALS.cleanShuffleDependencies"? Without checkpoint, we can still call "cleanShuffleDependencies".